website/docs/reference/widgets/table/README.md

```diff
@@ -60,183 +114,231 @@ Now, the table widget should be populated with the data coming from the query.
+
+<VideoEmbed host="youtube" videoId="3ayioaw5uj8" title="How To Setup Server-Side Search For The Table Widget" caption="How To Setup Server-Side Search For The Table Widget"/>
+
+1. Call the API / query with the **onSearchTextChange** event in the table's properties pane.
+2. Pass the value of `Table1.searchText` within the API request / query.
 
-Drag a text widget onto the canvas and bind it to a selected table row of `Table_1`.\
-Open the property pane of the text widget and add the following snippet to its label:
+As a SQL string:
+```sql
+SELECT * FROM users WHERE name LIKE {{"%" + Table1.searchText + "%"}} ORDER BY id LIMIT 10;
+```
 
+As an API request with URL parameters:
 ```
-{{Table_1.selectedRow}}
+https://mock-api.appsmith.com/users?name={{Table1.searchText}}
 ```
 
-<VideoEmbed host="youtube" videoId="0Pl7p1sA1fY" title="selectedRow" caption="selectedRow"/>
+### Server side filtering
 
-The Text widget then displays the data of the row selected on the table.
-If no row is selected, selectedRow shows the column names with no data.
+Server side filtering uses the same principles as described in [server side searching](#server-side-searching): some term or value is sent to the database or API to be used to filter out unnecessary data from the requested dataset. In this case, you will choose a value that records must match in order to be returned in the query's response.
 
-![](/img/slectedRow\_default.png)
+Server-side filtering requires the use of another widget, such as a [Select widget](/reference/widgets/select/), which you can use to provide users a list of supported filters to choose from.
 
-**Displaying column value using selectedRow**
+1. Drag a select widget to the canvas and add options that you might use to filter your data
+2. Set the table widget's **onOptionChange** event to call your API / query 
+3. Pass the Select widget's `selectedOptionValue` within the API request / query string
 
-If you want to fetch the data from a column in the row selected on the table, add the column's name after the selected row. You can refer to the code snippet below:
+As a SQL query:
+```sql
+SELECT * FROM users WHERE gender = {{genderDropdown.selectedOptionValue}};
+```
 
+As an API request with URL parameters:
 ```
-{{<table_name>.selectedRow.<column_name>}}
+https://mock-api.appsmith.com/users?gender={{genderDropdown.selectedOptionValue}}
 ```
 
-This gives you the data for `<columnname>` of the selected row. Using this snippet in `Table_1`, suppose you want to bind the task of the selected row to the text widget. To do this open the property pane of the text widget and add the following to its label:
+## Work with table data
+
+These properties and concepts are used to access your table data with code and process the dataset so the values are correctly formatted in the table widget.
+
+#### selectedRow
```

Comment:
a general suggestion when explaining properties - give heading with some context around using the property. for eg **Accessing the selected row**  and directly dive into examples. Only provide videos where you feel users will not understand how to use a feature.